### PR TITLE
Add staging/velero with Minio backend support

### PR DESCRIPTION
**Pull request by shaneutt**
This helps us support an option for Velero to utilize minio as a backend by default when other providers are not configured.

Upstream: [stable/velero](https://github.com/helm/charts/stable/velero)

Blocks https://github.com/mesosphere/kubeaddons-configs/pull/146

Followup: https://jira.mesosphere.com/browse/DCOS-56867

### CODE REVIEW

File: staging/velero/README.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# Velero-server
+
+This helm chart installs Velero version v1.0.0
+https://github.com/heptio/velero/tree/v1.0.0
+
```
Comment (alejandroEsc) [inline staging/velero/README.md]: delete one line

File: staging/velero/README.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+`metrics.enabled` | Set this to `true` to enable exporting Prometheus monitoring metrics | `false`
+`metrics.scrapeInterval` | Scrape interval for the Prometheus ServiceMonitor | `30s`
+`metrics.serviceMonitor.enabled` | Set this to `true` to create ServiceMonitor for Prometheus operator | `false`
+`metrics.serviceMonitor.additionalLabels` | Additional labels that can be used so ServiceMonitor will be discovered by Prometheus | `{}`
+`schedules` | A dict of schedules | `{}`
+`restic.podVolumePath` | Location of pod volumes on the host | `/var/lib/kubelet/pods`
+`restic.privileged` | Whether restic should run as a privileged pod. Only necessary in special cases (SELinux) | `false`
+`restic.resources` | Restic DaemonSet resource requests and limits | `{}`
+`configMaps` | Velero ConfigMaps | `[]`
+`minioBackend` | Enable Minio Backend | `false`
```
Comment (alejandroEsc) [inline staging/velero/README.md]: should be placed in alphabetical order? hard to tell here can you check?

**Comment by joejulian** [comment]
Please reference the upstream PR in the description.

**Comment by shaneutt** [comment]
updated @joejulian

File: staging/velero/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+    {{- with .Values.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+{{- if or .Values.initContainers .Values.minioBackend }}
+      initContainers:
+        {{- with .Values.minioBackend }}
```
Comment (alejandroEsc) [inline staging/velero/templates/deployment.yaml]: are these minio related pieces? can we include them in a values entry so that they are customizable, such as the image to use etc..

File: staging/velero/templates/miniobackend.yaml [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  version: v1beta1
+  names:
+    kind: MinIOInstance
+    plural: minioinstances
+  scope: Namespaced
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: minio-operator-role
```
Comment (alejandroEsc) [inline staging/velero/templates/miniobackend.yaml]: and here? also heritage info?

File: staging/velero/templates/miniobackend.yaml [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+  - min.io
+  resources:
+  - "*"
+  verbs:
+  - "*"
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: minio-operator-sa
```
Comment (alejandroEsc) [inline staging/velero/templates/miniobackend.yaml]: should we include release name here?

File: staging/velero/README.md (hunk 1 above)
Comment (shaneutt) [reply staging/velero/README.md]: So this is from the source repository, when I make an upstream PR for this I would rather not include changes like this yeah?

File: staging/velero/README.md (hunk 2 above)
Comment (shaneutt) [reply staging/velero/README.md]: Again, this is how upstream had it. Need to touch as little as possible.

File: staging/velero/templates/deployment.yaml (hunk 3 above)
Comment (shaneutt) [reply staging/velero/templates/deployment.yaml]: It is included in values as `minioBackend: false`

File: staging/velero/templates/miniobackend.yaml (hunk 5 above)
Comment (shaneutt) [reply staging/velero/templates/miniobackend.yaml]: It wont be necessary for our immediate purposes but may be something to note for the follow up for this in https://jira.mesosphere.com/browse/DCOS-56867

File: staging/velero/templates/miniobackend.yaml (hunk 4 above)
Comment (shaneutt) [reply staging/velero/templates/miniobackend.yaml]: Yep, good call, forgot. 49f68ee2e0efabb3714e74604b6f866d897316f0 takes care of this.

**Comment by shaneutt** [comment]
Done @joejulian

**Comment by joejulian** [comment]
For posterity, the intent of the /staging path is for temporary use while awaiting an upstream merge. We are not intending to fork upstream charts and maintain them ourselves.

I am pointing out that this PR without an associated upstream PR is creating tech debt that we have no specific plan to repay and I am against that.

File: staging/velero/README.md (hunk 1 above)
Comment (shaneutt) [reply staging/velero/README.md]: I've split the commits to better show the old vs new, let me know if this is still an issue.

File: staging/velero/README.md (hunk 2 above)
Comment (shaneutt) [reply staging/velero/README.md]: I've split the commits to better show the old vs new, let me know if this is still an issue.

**Comment by shaneutt** [comment]
@joejulian actually, the plan is detailed in https://jira.mesosphere.com/browse/DCOS-56867. I've added that to the desc now.
